### PR TITLE
Fixing CI scripts

### PR DESCRIPTION
**Pull request by adamrushy**
### 🎯 Goal

The goal is to automate our full release pipeline as per the previous tickets. We discovered some issues on the latest release and this should fix those.

### 🛠 Implementation

We have extracted the logic for tweaking the XCFrameworks and we'll look at doing this after this release.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change follows zero ⚠️ policy (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -13,8 +13,34 @@ jobs:
...
+          restore-keys: ${{ runner.os }}-mint-
+      - name: Extract version from branch name (for release branches)
         if: startsWith(github.event.pull_request.head.ref, 'release/')
-              BRANCH_NAME="${{ github.event.pull_request.head.ref }}"
-              VERSION=${BRANCH_NAME#release/}
-        run: bundle exec fastlane publish_release version:$VERSION
+        run: |
+          BRANCH_NAME="${{ github.event.pull_request.head.ref }}"
+          VERSION=${BRANCH_NAME#release/}
+          echo "RELEASE_VERSION=$VERSION" >> $GITHUB_ENV
```
Comment (adamrushy) [inline .github/workflows/publish-release.yml:42]: This is the fix here, setting as a Github Env variable means we can use it further down.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -110,6 +113,33 @@ lane :publish_release do |options|
...
+  # update_frameworks
+
+  # Create Github Release
+  github_release = set_github_release(
+                     repository_name: "GetStream/stream-chat-swift",
+                     api_token: ENV["GITHUB_TOKEN"],
+                     name: version,
+                     tag_name: version,
+                     description: changes
+                     # upload_assets: ["Products/StreamChat.zip", "Products/StreamChatUI.zip", "Products/StreamChat-All.zip"]
```
Comment (adamrushy) [inline fastlane/Fastfile:126]: These will be added manually _for now_

File: fastlane/Fastfile [hunk 3]
```diff
@@ -110,6 +113,33 @@ lane :publish_release do |options|
...
+  # Set tag
+  sh("git tag #{version}")
+
+  # Push tag to remote
+  push_to_git_remote(tags: true)
+
+  UI.success("Github release was created, please visit #{github_release["url"]} to see it")
+end
+
+desc "Update XCFrameworks and submit to the SPM repository"
```
Comment (adamrushy) [inline fastlane/Fastfile:141]: This is extracted logic to deal with the frameworks as part of the release. Github Actions struggles to deal with multiple repo's in the same action, I am likely to make this a new workflow.

File: fastlane/Fastfile [hunk 4]
```diff
@@ -31,6 +31,9 @@ end
 
 desc "Start a new release"
 lane :release do |options|
+  # Ensure we are on the main branch
+  ensure_git_branch(branch: 'main')
```
Comment (polqf) [inline fastlane/Fastfile:35]: Why is this back? A release will start from develop now, in most of the cases.

File: fastlane/Fastfile (hunk 2 above)
Comment (polqf) [reply fastlane/Fastfile:126]: Could we add a comment in `update_frameworks` so that we remember that we should be updating the release there?

**Pull request closed without merging**